freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/WorkloadCredentialUpdate.java

```diff
@@ -0,0 +1,35 @@
+package com.sequenceiq.freeipa.service.freeipa.user.model;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class WorkloadCredentialUpdate {
+
+    private final String username;
+
+    private final String userCrn;
+
+    private final WorkloadCredential workloadCredential;
+
+    public WorkloadCredentialUpdate(String username, String userCrn, WorkloadCredential workloadCredential) {
+        checkArgument(!StringUtils.isBlank(username), "username must not be blank");
+        this.username = username;
+        checkArgument(!StringUtils.isBlank(userCrn), "user CRN must not be blank");
```

Comment:
in my opinion it would be better to have all the validations first then the assignments

Reply:
Done.